internal/util/csiconfig.go

```diff
@@ -154,10 +154,54 @@ func GetMonsAndClusterID(ctx context.Context, clusterID string, checkClusterIDMa
 
 // GetClusterID fetches clusterID from given options map.
 func GetClusterID(options map[string]string) (string, error) {
-	clusterID, ok := options[clusterIDKey]
+	clusterID, ok := options[ClusterIDKey]
 	if !ok {
 		return "", ErrClusterIDNotSet
 	}
 
 	return clusterID, nil
 }
+
+// GetClusterIDFromMon will be called with a mon string to fetch
+// clusterID based on the passed in mon string. If passed in 'mon'
+// string has been found in the config the clusterID is returned,
+// else error.
+func GetClusterIDFromMon(mon string) (string, error) {
+	clusterID, err := readClusterInfoWithMon(CsiConfigFile, mon)
+	if err != nil {
+		return "", err
+	}
+
+	return clusterID, nil
+}
+
+func readClusterInfoWithMon(pathToConfig, mon string) (string, error) {
+	var config []ClusterInfo
+
+	// #nosec
+	content, err := ioutil.ReadFile(pathToConfig)
+	if err != nil {
+		err = fmt.Errorf("error fetching configuration file %q: %w", pathToConfig, err)
+
+		return "", err
+	}
+
+	err = json.Unmarshal(content, &config)
+	if err != nil {
+		return "", fmt.Errorf("unmarshal failed (%w), raw buffer response: %s",
+			err, string(content))
+	}
+
+	for _, cluster := range config {
+		if cluster.RadosNamespace != "" {
```

Comment:
add  a comment why we are continuing in this case.

Reply:
sure, done .. thanks..